src/parser/jobs/rpr/modules/HarvestMoon.tsx

```diff
@@ -35,37 +36,60 @@ export class HarvestMoon extends Analyser {
 		this.addEventHook('complete', this.onComplete)
 	}
 
-	private canChargeMoon(inputWindow: {start: number, end: number}): boolean {
-		const ADJUSTED_CAST_TIME = this.data.actions.SOULSOW.castTime + SOULSOW_BUFFER
+	private getExpectedUses(): number {
+		return this.invulWindows()
+			.filter(this.canChargeMoon.bind(this))
+			.length + 1
+	}
 
-		// get the earliest unable to act window that falls within the provided inputWindow
-		const unableToActWindow = this.unableToAct.getWindows(inputWindow)[0]
+	private invulWindows(): Window[] {
+		return this.invulnerability.getWindows().filter(this.longerThanChargeTime.bind(this))
+	}
 
-		// the window will be undefined if there are no unable to act windows left before the end of the inputWindow
-		if (unableToActWindow == null) {
-			// True if there are ADJUSTED_CAST_TIME milliseconds or more remaining in the inputWindow
-			return inputWindow.end - inputWindow.start >= ADJUSTED_CAST_TIME
-		}
+	private canChargeMoon(invulWindow: Window): boolean {
+		return this.unableToActWindows(invulWindow)
+			.reduce(this.toActableWindows, [] as Window[])
+			.some(this.longerThanChargeTime.bind(this))
+	}
 
-		// return true if there are ADJUSTED_CAST_TIME milliseconds between the beginning of the window being checked and the beginning of an unableToAct window
-		if (unableToActWindow.start - inputWindow.start >= ADJUSTED_CAST_TIME) {
-			return true
+	// Reduce function for calculating interstitial windows that assumes that there are at least 2 windows.
+	private toActableWindows(acc: Window[], window: Window, idx: number, arr: Window[]): Window[] {
```

Comment:
Ack generally prefers that you not abbreviate or shorten your variable / parameter names.

Reply:
Fixed this with more expressive variable names.